crates/miden-testing/tests/auth/multisig.rs

```diff
@@ -0,0 +1,414 @@
+        .with_component(BasicWallet)
+        .account_type(AccountType::RegularAccountUpdatableCode)
+        .storage_mode(AccountStorageMode::Public)
+        .with_assets(vec![FungibleAsset::mock(asset_amount)])
+        .build_existing()?;
+
+    Ok(multisig_account)
+}
+
+// ================================================================================================
+// TESTS
+// ================================================================================================
+
+#[tokio::test]
+async fn test_multisig() -> anyhow::Result<()> {
+    // ROLES
+    // - 2 Approvers (multisig signers)
+    // - 1 Multisig Contract
+
+    // Setup keys and authenticators
+    let (_secret_keys, public_keys, authenticators) = setup_keys_and_authenticators(2, 2)?;
+
+    // Create multisig account
+    let multisig_starting_balance = 10u64;
+    let mut multisig_account = create_multisig_account(2, &public_keys, multisig_starting_balance)?;
+
+    let mut mock_chain = MockChainBuilder::with_accounts([multisig_account.clone()])
+        .unwrap()
+        .build()
+        .unwrap();
+
+    // Create input note for the transaction
+    let input_note_asset = FungibleAsset::mock(5);
+    let input_note = create_p2id_note(
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE.try_into().unwrap(),
+        multisig_account.id(),
+        vec![input_note_asset],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new(Word::from([3u32; 4])),
+    )?;
+
+    mock_chain.add_pending_note(OutputNote::Full(input_note.clone()));
+    mock_chain.prove_next_block()?;
+
+    // Create output note for the transaction
+    let output_note_asset = FungibleAsset::mock(7);
+    let output_note = create_p2id_note(
+        multisig_account.id(),
+        ACCOUNT_ID_REGULAR_PUBLIC_ACCOUNT_UPDATABLE_CODE.try_into().unwrap(),
+        vec![output_note_asset],
+        NoteType::Public,
+        Default::default(),
+        &mut RpoRandomCoin::new(Word::from([3u32; 4])),
+    )?;
+
+    let tx_script = ScriptBuilder::with_kernel_library()?
+        .compile_tx_script(format!(
+            "
+            use.miden::tx
+            begin
+                push.{recipient}
+                push.{note_execution_hint}
+                push.{note_type}
+                push.0              # aux
+                push.{tag}
+                call.tx::create_note
+
+                push.{asset}
+                call.::miden::contracts::wallets::basic::move_asset_to_note
+                dropw dropw dropw dropw
+            end
+            ",
+            recipient = word_to_masm_push_string(&output_note.recipient().digest()),
+            note_type = NoteType::Public as u8,
+            tag = Felt::from(output_note.metadata().tag()),
+            asset = word_to_masm_push_string(&output_note_asset.into()),
+            note_execution_hint = Felt::from(output_note.metadata().execution_hint()),
+        ))
+        .unwrap();
```

Comment:
Creating an output note is just done to make the transaction "more interesting", right? But that code is not really under test here from what I can tell. If so, it would be nice to make this a bit more concise so the test gets to the interesting part more quickly. Could you check if we could use `MockChainBuilder::add_spawn_note` with the created output note, to have both an input and an output note in the transaction, but without the manual note creation?